flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/GenericHiveMetastoreCatalogUtil.java

```diff
@@ -118,18 +123,26 @@ public static Table createHiveTable(ObjectPath tablePath, CatalogBaseTable table
 				hiveTable.setPartitionKeys(new ArrayList<>());
 			}
 
-			hiveTable.setSd(sd);
+			hiveTable.setTableType(TableType.EXTERNAL_TABLE.name());
 		} else {
```

Comment:
I would suggest to write `else if (table instanceof CatalogView) {` here, and throw an exception in another `else`, to avoid we further add some more table type and the condition will be mis-matched here.